### ISSUE TITLE
[BUG] Upgrade from v1.2.x to master-head fails if backupTarget exists

### ISSUE THREAD

**Issue by khushboo-rancher** [longhorn/longhorn#3677]
## Describe the bug

The upgrade from v1.2.x-head to master-head fails. Admission hooks fail to start which results in failure of longhorn-manager pods.

## To Reproduce

1. Deploy Longhorn v1.2.x-head `02/28/2022`
2. Set some S3 backup target.
3. Create some backups.
4. Upgrade the Longhorn to master-head, Admission hooks fails to come up healthy.
```
2022/03/01 01:38:30 proto: duplicate proto type registered: VersionResponse
--
Mon, Feb 28 2022 5:38:30 pm | time="2022-03-01T01:38:30Z" level=info msg="Starting longhorn admission webhook server"
Mon, Feb 28 2022 5:38:30 pm | W0301 01:38:30.154415 1 client_config.go:552] Neither --kubeconfig nor --master was specified. Using the inClusterConfig. This might not work.
Mon, Feb 28 2022 5:38:30 pm | I0301 01:38:30.157467 1 shared_informer.go:223] Waiting for caches to sync for longhorn datastore
Mon, Feb 28 2022 5:38:30 pm | E0301 01:38:30.244825 1 reflector.go:178] github.com/longhorn/longhorn-manager/k8s/pkg/client/informers/externalversions/factory.go:117: Failed to list *v1beta2.BackupTarget: v1beta2.BackupTargetList.Items: []v1beta2.BackupTarget: v1beta2.BackupTarget.Status: v1beta2.BackupTargetStatus.Conditions: []v1beta2.Condition: decode slice: expect [ or n, but found {, error found in #10 byte of ...\|ditions":{},"lastSyn\|..., bigger context ...\|38:05Z"},"status":{"available":true,"conditions":{},"lastSyncedAt":"2022-03-01T01:38:05Z","ownerID":\|...
Mon, Feb 28 2022 5:38:31 pm | E0301 01:38:31.367725 1 reflector.go:178] github.com/longhorn/longhorn-manager/k8s/pkg/client/informers/externalversions/factory.go:117: Failed to list *v1beta2.BackupTarget: v1beta2.BackupTargetList.Items: []v1beta2.BackupTarget: v1beta2.BackupTarget.Status: v1beta2.BackupTargetStatus.Conditions: []v1beta2.Condition: decode slice: expect [ or n, but found {, error found in #10 byte of ...\|ditions":{},"lastSyn\|..., bigger context ...\|38:05Z"},"status":{"available":true,"conditions":{},"lastSyncedAt":"2022-03-01T01:38:05Z","ownerID":\|...
```

## Expected behavior

Admission hooks and longhorn-manager pods should come up healthy on upgrading the Longhorn.

## Log or Support bundle

[longhorn-admission-webhook-57c4747b5d-wprrb_longhorn-admission-webhook.log](https://github.com/longhorn/longhorn/files/8158177/longhorn-admission-webhook-57c4747b5d-wprrb_longhorn-admission-webhook.log)

## Environment

 - Longhorn version: master-head `02/28/2022`
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: Ubuntu 20.04
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 3

**Comment by derekbit** [comment]
@khushboo-rancher 
Was the backup job in progress?

**Comment by derekbit** [comment]
@khushboo-rancher 
While the backup job was in progress, the upgrade was also successful.
Could you provide me the support bundle?
I think the root cause is same as https://github.com/longhorn/longhorn/issues/3631.

**Comment by khushboo-rancher** [comment]
There was no backup in progress in my set up. I took below steps before upgrading.
1. I deployed v1.2.3 Longhorn in the cluster.
2. I had some volumes (RWX and RWO) with some random data written into it.
3. I took some CSI snapshots of the volumes on a S3 backupStore.
4. Upgraded it to v1.2.x-head.
5. Took some more CSI snapshot.
6. Now, upgraded to master-head where the issue occurred.

I couldn't generate the support bundle as the Longhorn-system was not fully up. Webhooks prevented Longhorn-manager to be deployed properly which results Longhorn deployer not to come up as well. So, overall my Longhorn deployment was stuck.
